engine/src/lib.rs

```diff
@@ -317,6 +318,29 @@ mod contract {
         );
     }
 
+    /// Updates the bytecode for user's router contracts created by the engine.
+    /// These contracts are where cross-contract calls initiated by the EVM precompile
+    /// will be sent from.
+    #[no_mangle]
+    pub extern "C" fn factory_update() {
+        let mut io = Runtime;
+        let state = engine::get_state(&io).sdk_unwrap();
+        require_owner_only(&state, &io.predecessor_account_id());
+        let bytes = io.read_input().to_vec();
+        let router_bytecode = crate::xcc::RouterCode(bytes);
+        crate::xcc::update_router_code(&mut io, &router_bytecode);
+    }
+
+    /// Updates the bytecode version for the given account. This is only called as a callback
+    /// when a new version of the router contract is deployed to an account.
+    #[no_mangle]
+    pub extern "C" fn factory_update_address_version() {
+        let mut io = Runtime;
+        io.assert_private_call().sdk_unwrap();
+        let args: crate::xcc::AddressVersionUpdateArgs = io.read_input_borsh().sdk_unwrap();
+        crate::xcc::set_code_version_of_address(&mut io, &args.address, args.version);
+    }
```

Comment:
Add this two methods to `TransactionKind` in `engine-standalone-storage`.

Outside of the scope of this PR: It would be great to enforce that every public method from the engine has an entry in the `TransactionKind` enum at compile level. It could be enforced in a test.

Reply:
Done in https://github.com/aurora-is-near/aurora-engine/pull/560/commits/1dfe17bde9b10f1223af748b948e42c885881448